src/mlpack/methods/preprocess/preprocess_describe_main.cpp

```diff
@@ -0,0 +1,231 @@
+
+using namespace mlpack;
+using namespace mlpack::data;
+using namespace std;
+using namespace boost;
+
+PROGRAM_INFO("Descriptive Statistics", "This utility takes a dataset and "
+    "prints out the descriptive statistics of the data. Descriptive statistics "
+    "is the discipline of quantitatively describing the main features of a "
+    "collection of information, or the quantitative description itself. The "
+    "program does not modify the original file, but instead prints out the "
+    "statistics to the console. The printed result will look like a table."
+    "\n\n"
+    "Optionally, width and precision of the output can be adjusted by a user "
+    "using the --width (-w) and --precision (-p). A user can also select a "
+    "specific dimension to analyize if he or she has too many dimensions."
+    "--population (-P) is a flag which can be used when the user wants the "
+    "dataset to be considered as a population. Otherwise, the dataset will "
+    "be considered as a sample."
+    "\n\n"
+    "So, a simple example where we want to print out statistical facts about "
+    "dataset.csv, and keep the default settings, we could run"
+    "\n\n"
+    "$ mlpack_preprocess_describe -i dataset.csv -v"
+    "\n\n"
+    "If we want to customize the width to 10 and precision to 5 and consider "
+    "the dataset as a population, we could run"
+    "\n\n"
+    "$ mlpack_preprocess_describe -i dataset.csv -w 10 -p 5 -P -v");
+
+// Define parameters for data.
+PARAM_STRING_IN_REQ("input_file", "File containing data,", "i");
+PARAM_INT_IN("dimension", "Dimension of the data. Use this to specify a "
+    "dimension", "d", 0);
+PARAM_INT_IN("precision", "Precision of the output statistics.", "p", 4);
+PARAM_INT_IN("width", "Width of the output table.", "w", 8);
+PARAM_FLAG("population", "If specified, the program will calculate statistics "
+    "assuming the dataset is the population. By default, the program will "
+    "assume the dataset as a sample.", "P");
+
+/**
+* Calculates the sum of deviations to the Nth Power
+*
+* @param input Vector that captures a dimension of a dataset
+* @param rowMean Mean of the given vector.
+* @return sum of nth power deviations
+*/
+double SumNthPowerDeviations(const arma::rowvec& input,
+    const double& rowMean,
+    const size_t Nth) // Degree of Power
+{
+  double sum = 0;
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+    sum += pow(input(i) - rowMean, Nth);
+  }
+  return sum;
+}
+/**
+ * Calculates Skewness of the given vector.
+ *
+ * @param input Vector that captures a dimension of a dataset
+ * @param rowStd Standard Deviation of the given vector.
+ * @param rowMean Mean of the given vector.
+ * @return Skewness of the given vector.
+ */
+double Skewness(const arma::rowvec& input,
+    const double& rowStd,
+    const double& rowMean,
+    const bool population)
+{
+  double skewness = 0;
+  double S3 = pow(rowStd, 3);
+  double M3 = SumNthPowerDeviations(input, rowMean, 3);
```

Comment:
I guess M3 mean moment 3?If so, I think write it as
`M3/n` maybe can express your intention better